Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/Octree_wrapper.h

```diff
@@ -0,0 +1,625 @@
+      {
+        Uniform_coords vuc = vertex_uniform_coordinates(node, i);
+        const auto lex = lex_index(vuc[0], vuc[1], vuc[2], max_depth_);
+        leaf_vertices_set.insert(lex);
+      }
+
+      // write all leaf edges in a set
+      const auto& coords_global = node.global_coordinates();
+      const auto& depth = node.depth();
+      const auto& df = depth_factor(node.depth());
+      for(const auto& edge_voxels : Tables::edge_to_voxel_neighbor)
+      {
+        bool are_all_voxels_leafs = true;
+        for(const auto& node_ijk : edge_voxels)
+        {
+          const std::size_t x = coords_uniform[0] + df * node_ijk[0];
+          const std::size_t y = coords_uniform[1] + df * node_ijk[1];
+          const std::size_t z = coords_uniform[2] + df * node_ijk[2];
+          // check for overflow / ignore edges on boundary
+          if(x >= dim_ || y >= dim_ || z >= dim_)
+          {
+            are_all_voxels_leafs = false;
+            break;
+          }
+
+          const Node n = get_node(x, y, z);
+          if(n.depth() > depth)
+          {
+            are_all_voxels_leafs = false;
+            break;
+          }
+        }
+
+        if(are_all_voxels_leafs)
+        {
+          // add to leaf edge set
+          std::size_t e_gl = e_glIndex(edge_voxels[0][3],
+                                       coords_global[0], coords_global[1], coords_global[2],
+                                       depth);
+          leaf_edges_set.insert({e_gl, depth});
+        }
+      }
+    }
+
+    leaf_voxels_ = std::vector<Voxel_handle>{leaf_voxels_set.begin(), leaf_voxels_set.end()};
+    leaf_edges_ = std::vector<Edge_handle>{leaf_edges_set.begin(), leaf_edges_set.end()};
+    leaf_vertices_ = std::vector<Vertex_handle>{leaf_vertices_set.begin(), leaf_vertices_set.end()};
+  }
+
+  const Octree& octree() const
+  {
+    return octree_;
+  }
+
+  const Geom_traits& geom_traits() const
+  {
+    return gt_;
+  }
+
+  std::size_t dim() const
+  {
+    return dim_;
+  }
+
+  FT hx() const
+  {
+    return hx_;
+  }
+
+  FT offset_x() const
+  {
+    return offset_x_;
+  }
+
+  FT offset_y() const
+  {
+    return offset_y_;
+  }
+
+  FT offset_z() const
+  {
+    return offset_z_;
+  }
+
+  std::size_t max_depth() const
+  {
+    return max_depth_;
+  }
+
+  const std::vector<Edge_handle>& leaf_edges() const
+  {
+    return leaf_edges_;
+  }
+
+  const std::vector<Vertex_handle>& leaf_vertices() const
+  {
+    return leaf_vertices_;
+  }
+
+  const std::vector<Voxel_handle>& leaf_voxels() const
+  {
+    return leaf_voxels_;
+  }
+
+  std::size_t depth_factor(const std::size_t& depth) const
+  {
+    return std::size_t(1) << (max_depth_ - depth);
+  }
+
+  Uniform_coords uniform_coordinates(const Node& node) const
+  {
+    auto coords = node.global_coordinates();
+    const std::size_t df = depth_factor(node.depth());
+    for(int i=0; i<Node::Dimension::value; ++i)
+      coords[i] *= df;
```

Comment:
In my  local compilation the elements of `coord[i]` are of type `uint_32` so we get a warning. Would it make sense to make the return type of `depth_factor()` smaller?